packages/huddle01-interfaces/src/AudioSpacesInterface.ts

```diff
@@ -0,0 +1,58 @@
+import { IPeer } from '@huddle01/web-core';
+
+export type TRoomControls = 'muteEveryone';
+
+export default abstract class AudioSpacesInterface {
```

Comment:
Great abstraction.

Few things to note:

# Specific to this PR

1. There are some functions common to both AudioSpacesInterface and VideoMeetInterface ex: Lobby Functions, Hosting and Control functions. They can be put in a super class `SpacesAbstract`. The `AudioSpacesInterface` and `VideoMeetInterface` extend from that class, and additionally add any other abstract functions needed for their specific functionality.

# For overall project hygiene

Right now, the packages have diverged. It was supposed to be a common `messaging` package that implements the audio, video, text and 1:1, 1:N communication. However, Push's Video implementation is a separate package and Huddle01's a separate. So, we must make a one time effort to merge these packages together. The goal is to have

1. Plug and Play functionality for developers. Switching b/w Push, Huddle01 and other providers of such communication should be just config change.
2. Standard abstractions to enable more protocols to be added seamlessly.
3. A great example for others to look at extending dApp-SDK classes.

These changes are out of scope of this PR.

We will pick this up as a separate effort, and update it later.